File: src/frontend/src/app/core/models/field-of-study/field-of-study.ts
import { CourseType } from "../../enums/field-of-study/course-type.enum";
import { DegreeLevel } from "../../enums/field-of-study/degree-level.enum";
import { MainLanguage } from "../../enums/field-of-study/main-language.enum";
import { StudiesProfile } from "../../enums/field-of-study/studies-profile.enum";
import { NonVersionedModelBase } from "../non-versioned-model-base";
import { User } from "../user/user";
import { Specialization } from "./specialization";

export class FieldOfStudy extends NonVersionedModelBase {
    name?: string;
    level: DegreeLevel = DegreeLevel.FirstLevel;
    profile: StudiesProfile = StudiesProfile.Academic;
    branchOfScience?: string;
    discipline?: string;
    faculty = 'Informatyka i Zarządzanie';
    type: CourseType = CourseType.FullTime;
    language: MainLanguage = MainLanguage.Polish;
    supervisor?: User;
    specializations: Specialization[] = [];
}

File: src/backend/SyllabusManager.Data/Models/FieldOfStudies/FieldOfStudy.cs
using SyllabusManager.Data.Attributes;
using SyllabusManager.Data.Enums.FieldOfStudies;
using SyllabusManager.Data.Models.User;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SyllabusManager.Data.Models.FieldOfStudies
{
    /// <summary>
    /// Kierunek studiów
    /// </summary>
    public class FieldOfStudy : NonVersionedModelBase
    {

        [Required]
        [PdfName("Nazwa")]
        public string Name { get; set; }
        [PdfName("Poziom")]
        public DegreeLevel Level { get; set; }
        [PdfName("Profil")]
        public StudiesProfile Profile { get; set; }
        [PdfName("Dziedzina nauki")]
        public string BranchOfScience { get; set; }
        [PdfName("Dyscyplina")]
        public string Discipline { get; set; }
        [Required]
        [PdfName("Wydział")]
        public string Faculty { get; set; } = "Informatyka i Zarządzanie";
        [PdfName("Typ kursu")]
        public CourseType Type { get; set; }
        [PdfName("Język")]
        public MainLanguage Language { get; set; }
        public SyllabusManagerUser Supervisor { get; set; }
        public List<Specialization> Specializations { get; set; } = new List<Specialization>();
    }
}